metric_learn/constraints.py

```diff
@@ -23,6 +23,7 @@ def __init__(self, partial_labels):
     self.num_points, = partial_labels.shape
     self.known_label_idx, = np.where(partial_labels >= 0)
```

Comment:
we are now storing redundant information into the class attributes. I think it would be better to simply save `partial_labels` and reconstruct the other arrays in the methods whenever needed